### DISCUSSION TITLE
HELP! Crashing in debug mode

### DISCUSSION THREAD

**Discussion by yoshymoo** [jrouwe/JoltPhysics#1271] [Q&A]
Hello! I am working on a school project to create a c++ 3d game engine and I'm using Jolt library for my physics system. 
However, my solution crashes whenever I close it in VS and this error pops up. how do I fix this? 

I'm not sure what I'm doing wrong, I am simply following the helloworld.cpp code :(  
![Screenshot 2024-09-20 175708](https://github.com/user-attachments/assets/bdbc1ea3-0da3-4a62-8173-5bb2a2d81ea3)
![Screenshot 2024-09-20 175626](https://github.com/user-attachments/assets/18cc9ee2-aa43-4a1c-bc5f-78602430b813)

**Comment by jrouwe** [comment]
Perhaps a mismatch in defines between your project and Jolt? You've provided very little information / no repro so I have no idea.

**Reply by yoshymoo** [reply]
thing is, i barely got anything started :( I only initialised the physics system which is up till here
`physics_system.Init(cMaxBodies, cNumBodyMutexes, cMaxBodyPairs, cMaxContactConstraints, broad_phase_layer_interface, object_vs_broadphase_layer_filter, object_vs_object_layer_filter);`
I can run my solution, but it crashes whenever I close the window.

Could it be how I link my external libs(jolt in this case) because mine is a single project and I am using non-static dll libs.

**Reply by jrouwe** [reply]
Possible. Jolt by default uses static libs (controlled by cmake option USE_STATIC_MSVC_RUNTIME_LIBRARY), although I think that usually ends up giving linking errors.

**Reply by yoshymoo** [reply]
I see. by any chance, is there any dll file available? because the sample project sln doesn't make one

**Reply by jrouwe** [reply]
cmake option BUILD_SHARED_LIBS
